aws/sdk/integration-tests/s3/tests/select-object-content.rs

```diff
@@ -0,0 +1,94 @@
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_sdk_s3::model::{
+    CompressionType, CsvInput, CsvOutput, ExpressionType, FileHeaderInfo, InputSerialization,
+    OutputSerialization, SelectObjectContentEventStream,
+};
+use aws_sdk_s3::{Client, Config, Credentials, Region};
+use smithy_client::dvr::{Event, ReplayingConnection};
+use std::error::Error as StdError;
+
+#[tokio::test]
+async fn test_success() {
+    let events: Vec<Event> =
+        serde_json::from_str(include_str!("select-object-content.json")).unwrap();
+    let replayer = ReplayingConnection::new(events);
+
+    let region = Region::from_static("us-east-2");
+    let credentials = Credentials::from_keys("test", "test", None);
+    let config = Config::builder()
+        .region(region)
+        .credentials_provider(credentials)
+        .build();
+    let client = Client::from_conf_conn(config, replayer.clone());
+
+    let mut output = client
+        .select_object_content()
+        .bucket("aws-rust-sdk")
+        .key("sample_data.csv")
+        .expression_type(ExpressionType::Sql)
+        .expression("SELECT * FROM s3object s WHERE s.\"Name\" = 'Jane'")
+        .input_serialization(
+            InputSerialization::builder()
+                .csv(
+                    CsvInput::builder()
+                        .file_header_info(FileHeaderInfo::Use)
+                        .build(),
+                )
+                .compression_type(CompressionType::None)
+                .build(),
+        )
+        .output_serialization(
+            OutputSerialization::builder()
+                .csv(CsvOutput::builder().build())
+                .build(),
+        )
+        .send()
+        .await
+        .unwrap();
+
+    let mut received = Vec::new();
+    while let Some(event) = output.payload.recv().await.unwrap() {
+        match event {
+            SelectObjectContentEventStream::Records(records) => {
+                received.push(
+                    std::str::from_utf8(records.payload.as_ref().unwrap().as_ref())
+                        .unwrap()
+                        .trim()
+                        .to_string(),
+                );
+            }
+            SelectObjectContentEventStream::Stats(stats) => {
+                let stats = stats.details.unwrap();
+                received.push(format!(
+                    "scanned:{},processed:{},returned:{}",
+                    stats.bytes_scanned, stats.bytes_processed, stats.bytes_returned
+                ))
+            }
+            otherwise => panic!("unexpected message: {:?}", otherwise),
+        }
+    }
+    assert_eq!(
+        vec![
+            "Jane,(949) 555-6704,Chicago,Developer".to_string(),
+            "scanned:333,processed:333,returned:39".to_string()
+        ],
+        received
+    );
+
+    // Validate the requests
+    replayer
+        .validate(&["content-type", "content-length"], validate_body)
```

Comment:
how does this work without timesource pinning?

Reply:
S3 Select isn't a bidirectional stream, so we only get events back in the response. So that means only the initial request is being signed, and we're not validating the signature headers here.